Modules/Loadable/Markups/VTKWidgets/vtkSlicerMarkupsWidgetRepresentation3D.cxx

```diff
@@ -627,15 +677,34 @@ void vtkSlicerMarkupsWidgetRepresentation3D::UpdateFromMRML(vtkMRMLNode* caller,
   for (int controlPointType = 0; controlPointType < NumberOfControlPointTypes; ++controlPointType)
     {
     double* color = this->GetWidgetColor(controlPointType);
+    double opacity = this->MarkupsDisplayNode->GetOpacity() * hierarchyOpacity;
 
     ControlPointsPipeline3D* controlPoints = this->GetControlPointsPipeline(controlPointType);
     controlPoints->Property->SetColor(color);
-    controlPoints->Property->SetOpacity(this->MarkupsDisplayNode->GetOpacity() * hierarchyOpacity);
+    controlPoints->Property->SetOpacity(opacity);
 
     controlPoints->TextProperty->SetColor(color);
-    controlPoints->TextProperty->SetOpacity(this->MarkupsDisplayNode->GetOpacity() * hierarchyOpacity);
+    controlPoints->TextProperty->SetOpacity(opacity);
     controlPoints->TextProperty->SetFontSize(static_cast<int>(5. * this->MarkupsDisplayNode->GetTextScale()));
 
+    controlPoints->OccludedProperty->SetColor(color);
+    controlPoints->OccludedTextProperty->SetColor(color);
+    controlPoints->OccludedTextProperty->SetFontSize(controlPoints->TextProperty->GetFontSize());
+    if (this->MarkupsDisplayNode->GetOccludedVisibility() && this->MarkupsDisplayNode->GetOccludedOpacity() > 0.0)
+      {
+      // To prevent some rendering artifacts, and to ensure that the occluded actor does not block point visibility,
+      // the maximum opacity of the occluded actor is required to be almost, but not fully opaque.
+      double occludedOpacity =
+        std::min(0.99999999, this->MarkupsDisplayNode->GetOccludedOpacity() * opacity);
```

Comment:
This value is a bit risky, as 0.99999999 can only be stored in double, and not in float (in float it would be stored as 1.0, see https://www.h-schmidt.net/FloatConverter/IEEE754.html). Using 0.9999999 can be stored in single-precision float, but to avoid any numerical issues, we could use a very safe value such as 0.99 (that would be even non-opaque if stored in an unsigned char alpha channel).

Reply:
> https://www.h-schmidt.net/FloatConverter/IEEE754.html

This website is neat. Thanks for sharing

For future reference:

* using `0.99999999`
![image](https://user-images.githubusercontent.com/219043/91884322-d6e51700-ec53-11ea-986f-3f7cb354865c.png)

* using `0.99`
![image](https://user-images.githubusercontent.com/219043/91884360-e82e2380-ec53-11ea-8e90-06e2d2413a75.png)